src/wwwroot/controllers/TemplateCtrl.js

```diff
@@ -39,19 +39,25 @@
     function load(templateId) {
       $scope.loadInProgress = true;
 
-      templates.getTemplate(templateId)
-        .then(function (result) {
-          $scope.template.fromName = result.from_name;
-          $scope.template.fromEmail = result.from_email;
-          $scope.template.subject = result.subject;
-          $scope.template.content = result.body;
-          $scope.template.name = result.name;
-          $scope.template.id = result.id;
-        }).catch(function (error) {
-          // TODO: do something with the error
-        }).finally(function () {
-          $scope.loadInProgress = false;
-        });
+      templates.getTemplateBody(templateId)
```

Comment:
>  > Another thing, with this approach, we cannot take advantage of API links to get the body, I am not really sure, but we could move that responsability to service.

> Could you please explain me in more details? templates.getTemplateBody lives in template service.

I am talking about moving orchestration responsibility to the service. For example, adding a new method like _getTemplateWithBody_.